### PR TITLE
fix(react-textarea): Don't remove outline when filled and disabled and apply correct disabled color to text

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

image
image

## New Behavior

image

image

### CODE REVIEW

File: packages/react-components/react-textarea/src/components/Textarea/useTextareaStyles.styles.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ const useRootStyles = makeStyles({
     backgroundColor: tokens.colorTransparentBackground,
     ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStrokeDisabled),
     [`& > textarea`]: {
```
Comment (behowell) [inline packages/react-components/react-textarea/src/components/Textarea/useTextareaStyles.styles.ts:34]: Ideally we  should not be using a `& > textarea` selector here. Instead, these should be on a `disabled` class in `useTextareaStyles` and applied directly to the slot.

I know this isn't part of this change, so it doesn't need to block this PR, and could be fixed in a separate PR instead.

**Review by khmakoto — APPROVED** [review]

File: change/@fluentui-react-textarea-5bff6329-66e1-4f78-b54e-615647eb847c.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fix: Don't remove outline when filled and disabled and apply correct disabled color to text.",
```
Comment (behowell) [inline change/@fluentui-react-textarea-5bff6329-66e1-4f78-b54e-615647eb847c.json:3]: nit:
```suggestion
  "comment": "fix: When Textarea is disabled, make the text gray and add an outline for the filled appearance.",
```

**Review by behowell — APPROVED** [review]

File: packages/react-components/react-textarea/src/components/Textarea/useTextareaStyles.styles.ts (hunk 1 above)
Comment (sopranopillow) [reply packages/react-components/react-textarea/src/components/Textarea/useTextareaStyles.styles.ts:34]: Created an issue to track that work #27838

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/InfoButton-mount_base.html">16</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/InfoButton-mount.html">21</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/Avatar-mount_base.html">724</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/Avatar-mount.html">700</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/Button-mount_base.html">393</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/Button-mount.html">390</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/Field-mount_base.html">1250</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/Field-mount.html">1212</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/FluentProvider-mount_base.html">802</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/FluentProvider-mount.html">799</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/FluentProviderWithTheme-mount_base.html">112</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/FluentProviderWithTheme-mount.html">116</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">103</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">100</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">112</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">108</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/InfoButton-mount_base.html">16</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/InfoButton-mount.html">21</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/MakeStyles-mount_base.html">1190</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/MakeStyles-mount.html">1177</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/Persona-mount_base.html">1914</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/Persona-mount.html">1928</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/SpinButton-mount_base.html">1472</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27837/perf-test-react-components/SpinButton-mount.html">1433</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/79c08ce5bbf6387b2b18ba4c3d2d5681e0177d4b'>79c08ce5bbf6387b2b18ba4c3d2d5681e0177d4b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: apps/vr-tests-react-components/src/stories/Textarea.stories.tsx [hunk 3]
```diff
@@ -37,7 +37,11 @@ storiesOf('Textarea Converged', module)
       <Textarea aria-invalid appearance="filled-lighter" placeholder="Placeholder" />
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/Textarea.stories.tsx:37]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 9 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=27837&commitId=9885293745cdc15969b6279152b21b3238750e38&env=prod)

✅ There was 6 screenshots added, 0 screenshots removed, 1933 screenshots unchanged, 0 screenshots with different dimensions and 3 screenshots with visible difference.


  <details><summary><b>unknown</b> 9 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Textarea Converged.Disabled.default.chromium.png|114| Changed|
|Textarea Converged.Disabled.focused.chromium.png|114| Changed|
|Textarea Converged.Disabled.hover.chromium.png|114| Changed|
|Textarea Converged.Disabled + filled.default.chromium.png|0| Added|
|Textarea Converged.Disabled + filled.focused.chromium.png|0| Added|
|Textarea Converged.Disabled + filled.hover.chromium.png|0| Added|
|Textarea Converged.Disabled + placeholder.default.chromium.png|0| Added|
|Textarea Converged.Disabled + placeholder.focused.chromium.png|0| Added|
|Textarea Converged.Disabled + placeholder.hover.chromium.png|0| Added|

</details>


<div id="vrtCommentfluentuiv9"/>

**Pull request merged**